Tools/CISupport/ews-build/steps.py

```diff
@@ -67,6 +72,74 @@
 LAYOUT_TESTS_URL = '{}{}/blob/{}/LayoutTests/'.format(GITHUB_URL, GITHUB_PROJECTS[0], DEFAULT_BRANCH)
 
 
+class TwistedAdditions(object):
+    class Response(object):
+        def __init__(self, status_code, content=None, url=None, headers=None):
+            self.status_code = status_code
+            self.content = content or b''
+            self.url = url
+            self.headers = headers or {}
+
+        @property
+        def text(self):
+            return self.content.decode('utf-8')
+
+        def json(self):
+            return json.loads(self.text)
+
+    class Printer(protocol.Protocol):
+        def __init__(self, finished):
+            self.finished = finished
+            self.data = b''
+
+        def dataReceived(self, bytes):
+            self.data += bytes
+
+        def connectionLost(self, reason):
+            self.finished.callback(self.data)
+
+    @classmethod
+    @defer.inlineCallbacks
+    def request(cls, type=None, url=None, params=None, headers=None, logger=None, timeout=10):
+        logger = logger or (lambda _: None)
+        typ = type or b'GET'
+
+        if not url:
```

Comment:
since `url` is mandatory, let's not make it a optional parameter in `request(url=None)`